src/components/lazy-list/index.js

```diff
@@ -0,0 +1,327 @@
+import { useState, useLayoutEffect, useRef, useMemo, useCallback } from 'react';
+
+/**
+ * A component that renders only the children currently visible on screen.
+ * @param {function} height A `function(start, end)` returning the pixel height for any given range of items
+ * @param {number} total The total count of all items in the list
+ * @param {function} children A `function(props)` rendering the list and items (see `childProps`)
+ * @param {?number} [buffer=0.5] A number [0...1] as a % of the visible region to render additionally
+ * @param {?boolean} [lazy=true] Toggles the lazy functionality
+ * @param {?boolean} [dispose=false] Toggles disposing items when they lose visibility
+ * @param {?function} onChange Optional change callback
+ * @param {?function} container Optional, default scroll container is `element.offsetParent`
+ * @return {object} The rendered children
+ **/
+export default ({
+  height,
+  total,
+  children,
+  lazy = true,
+  dispose = false,
+  buffer = 0.5,
+  onChange,
+  container = element => element?.offsetParent
+}) => {
+  // Required feature checks
+  const supported = typeof window.IntersectionObserver !== 'undefined';
```

Comment:
What happens if it's not supported?

Reply:
The line below hopefully makes that clear:

```
// Active only if enabled by prop and features detected
const active = lazy && supported;
```

Then a few lines further down:

```
  // Skipped if not enabled or supported
  if (active) {
```

Basically, it will just 'pass through' and render the whole list. Shall I add that in the description?

Reply:
ah thanks 🙈! yeah not a bad idea to add that as a comment but it looks like it's already made fairly clear